shared/elements/gaia_grid/js/items/grid_item.js

```diff
@@ -514,7 +514,14 @@
 
         var nameEl = document.createElement('span');
         nameEl.className = 'title';
-        nameEl.textContent = this.name;
+
+        if (this.name && this.name.then) {
```

Comment:
I agree. I didn't want to alter the API without your permission, but I'd prefer to have asyncName and name or even have the name always be async?